File: CCPL/CCPL oct-17-2015/Earth.java
import java.io.*;
import java.math.*;
import java.util.StringTokenizer;

public class Earth {
    static double R = 6371009.00; 
    static double EPS = 1e-10;
    static double sphericalDistance(double latitudS,double longitudS,double latitudF,double longitudF){
       latitudS = Math.toRadians(latitudS);
       latitudF = Math.toRadians(latitudF);
       longitudS = Math.toRadians(longitudS);
       longitudF = Math.toRadians(longitudF);
    
       double deltaLongitud = longitudF - longitudS;
       double deltaLatitud = latitudF - latitudS;
       double a = Math.pow((Math.sin(deltaLatitud/2)),2) + Math.cos(latitudS) * Math.cos(latitudF) * Math.pow(Math.sin(deltaLongitud/2), 2);
       double c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1-a));
       double d = R * c;
       double e = Math.sqrt(2.*R*R *(1-Math.cos(c)));
       return Math.round(d-e+EPS);
    }
    
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        double latitudS, longitudS, latitudF, longitudF;
        String cases, lines;
        cases = br.readLine();
        StringTokenizer st;
        for(int i=0; i<Integer.valueOf(cases); i++){
            lines = br.readLine();
            st = new StringTokenizer(lines);
            latitudS = Double.valueOf(st.nextToken());
            longitudS = Double.valueOf(st.nextToken());
            latitudF = Double.valueOf(st.nextToken());
            longitudF = Double.valueOf(st.nextToken());
            
            double ans1 = sphericalDistance(latitudS, longitudS, latitudF, longitudF);
            
            System.out.println ((long)(0.5 + ans1));
            
        }
        
    }
}

File: CCPL/CCPL oct-17-2015/earth.cpp
#include <bits/stdc++.h>
#define pi 3.1415926535897932384626433832795
#define EPS 1e-10

using namespace std;

const double R = 6371009.00 ; // radius of earth / sphere

// great Circle in Sphere / points must be in radian / distance between 2 pnts on earth
double spherical_distance(double lat1,double lon1,double lat2,double lon2) {
       double dlon = lon2 - lon1;
       double dlat = lat2 - lat1;
       double a = pow((sin(dlat/2)),2) + cos(lat1) * cos(lat2) * pow(sin(dlon/2), 2);
       double c = 2 * atan2(sqrt(a), sqrt(1-a));
       double d = R * c;
       double e = sqrt(2.*R*R *(1-cos(c)));
       return round(d-e+EPS);
}

int main(){
	int t;
	cin>>t;
	while(t--){
		double ax,ay,bx,by;
		cin>>ax>>ay>>bx>>by;
		ax *= pi/180;
		ay *= pi/180;
		bx *= pi/180;
		by *= pi/180;
		long long ans = (long long)(0.5 + spherical_distance(ax,ay,bx,by));
		cout<<ans<<endl; 
	}
	return 0;
}

